core/src/main/java/org/apache/lucene/queryparser/classic/MapperQueryParser.java

```diff
@@ -779,7 +757,9 @@ private Query applySlop(Query q, int slop) {
                 builder.add(terms[i], positions[i]);
             }
             pq = builder.build();
-            pq.setBoost(q.getBoost());
+            //make sure that the boost hasn't been set beforehand, otherwise we'd lose it
+            assert q.getBoost() == 1f;
+            assert q instanceof BoostQuery == false;
```

Comment:
that works for me, let's just make sure that we have a unit test that sets both the slop and the boost on a phrase query